### ISSUE TITLE
Upgrade to RN 0.71.0

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#13217]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

This issue is a placeholder for work being done by @janicduplessis to help upgrade our RN fork to `0.71.0` after a large upstream refactor of the VirtualizedList component.

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by tgolen** [comment]
Is there anything I can do to help?

**Comment by roryabraham** [comment]
Hey @tgolen, @janicduplessis has a PR [here](https://github.com/Expensify/react-native/pull/36) that I haven't had time to review yet. If you want to learn more and dive in, you can help review that.

It has a lot of changes, but many of them are already well-tested and reviewed against older React Native versions. To understand all the custom changes in our fork that we already have today, you can look at [this diff](https://github.com/facebook/react-native/compare/v0.70.4...Expensify:react-native:Expensify-0.70.4-alpha.2) which compares the latest released version of our fork, `0.70.4-alpha.2`, with the upstream tag it's based on, `v.0.70.4`.

Understanding those changes, the main goal of https://github.com/Expensify/react-native/pull/36 is to port all of those changes over to a newer version of React Native, `0.71.0`, which introduced a [large refactor of the VirtualizedList component](https://github.com/facebook/react-native/commit/971599317b7bdf1152157206f9503a23ac8c4162).

The best way to test is by running `rn-tester`, which is a React Native application stored in the React Native repo under `packages/rn-tester`: https://github.com/Expensify/react-native/tree/main/packages/rn-tester

---

If you're feeling like there's too much context in all those changes, another way you can help out is by publishing a new version of our fork, `0.71.0-alpha.1` after the above PR is merged. So far only myself and @AndrewGable have done that, so it would be good to get another person with knowledge of how to do so, following [this SO](https://stackoverflow.com/c/expensify/questions/14023). We can then hire an external contributor to upgrade E/App to use that new version.

---

For a bit of context on the priority here, I don't know of any critical reason why we need to upgrade E/App to `0.71.0`, so while this is important to do so that we can upgrade when we need to, it isn't super urgent.

**Comment by tgolen** [comment]
Cool, that is a lot of stuff! I'd love to learn how to publish the fork, so let me jump in at that spot 👍

**Comment by roryabraham** [comment]
@tgolen that PR is merged, so we are ready to publish the fork now

**Comment by tgolen** [comment]
I've been working through several issues in this thread: https://expensify.slack.com/archives/C01GTK53T8Q/p1670610675815269

I got as far as trying to run the `rn-tester` app, but can't get it to build and run for some reason. I'm a bit stuck, and not sure what I need to do to get unstuck, so I'll wait to see if anyone has ideas in that thread.

**Comment by tgolen** [comment]
I haven't been able to get unstuck with this. @roryabraham I think we need to bring in the big guns to get this done? Can Janic do it?

**Comment by tgolen** [comment]
Szymon is going to try to run through the update path today using the most recent release-candidate from react native.

**Comment by tgolen** [comment]
Szymon is still putting some thoughts together about how we can get this deployed.

**Comment by Szymon20000** [comment]
It is released. Wasn't sure if you want me to work on it.
I have plenty of things at the moment on my plate but will do my best to work on it if there is no one else that could take it.
@janicduplessis Would you be able to help? 
The biggest issue with alpha version was that reanimated wasn't ready yet for the 0.71. However it is now.

**Comment by tgolen** [comment]
I honestly still feel woefully inadequate to perform the task :blush: and your in-depth understanding of all the moving pieces allows you to get unblocked much quicker than me when you run into problems.

My preference would be to have @Szymon20000 or @janicduplessis go through the process, write the process down, and then I can use that to perform the next update (putting the written down process through the test).

**Comment by tgolen** [comment]
I know we've gone around-and-around on this issue, but I think the latest discussion in Slack was that doing this maybe isn't such a high priority since:

1. We are not sure what benefits it gives us (it might be faster in some places and slower in others)
2. We are not sure when our third-party-libs will fully support this version
3. The upgrade is not blocking anything

I think because of that, I'm going to decrease this to even lower priority.

**Comment by janicduplessis** [comment]
I managed to publish the fork as a github release and updated the app to use it here https://github.com/Expensify/App/pull/14618. What is missing is publishing the fork on a release on expensify fork instead of my personal account. There also seems to be some CI failures that I will have to take a look at.

**Comment by tgolen** [comment]
Looks like this is still kinda on hold for now and there isn't a clear value proposition for getting this done right now.

**Comment by roryabraham** [comment]
Whoops, this is actually being done [over here](https://github.com/Expensify/App/pull/15130)
